### PR TITLE
fix improper offset with generics

### PR DESCRIPTION
**Pull request by ix0rai**
this fix is needed for enigma, since I want to make Quiltflower the default decompiler and this was the last issue to track down first. I also considered moving line 196 out of its else block as a fix, but that seems like it would just be a slower way to accomplish the same thing from a quick look.

fixes https://github.com/QuiltMC/enigma/issues/47, which was causing some generics to explode and look like this:
image

all credit for tracking this down goes to my brilliant friend @orifu :)

### CODE REVIEW

**Comment by jaskarth** [comment]
Looks nice! Can you add a unit test for this?

**Comment by ix0rai** [comment]
should I be worried about the currently failing tests?

**Comment by sschr15** [comment]
No - QF's unit tests are cursed at best currently

**Comment by ix0rai** [comment]
no idea if I did tests right, but I managed to make a test that fails without the change and succeeds with it

File: src/org/jetbrains/java/decompiler/util/TextBuffer.java [hunk 1]
```diff
@@ -187,7 +187,9 @@ public TextBuffer addGenericTypeTokens(int index, String text, VarType type) {
         if (gt.type == CodeConstants.TYPE_OBJECT) {
           addAllClassTokens(index + offset, name, gt.value);
         }
+
         addGenericTypeTokens(index + offset, name, gt);
+        offset += name.length();
```
Comment (ByMartrixx) [inline src/org/jetbrains/java/decompiler/util/TextBuffer.java:192]: Huh, what an oversight (it was my fault)

**Review by ByMartrixx — APPROVED** [review]

**Review by OroArmor — APPROVED** [review]

**Review by sschr15 — APPROVED** [review]

**Review by jaskarth — APPROVED** [review]
LGTM!

**Pull request merged**